Neos.Flow/Classes/Security/Policy/Roles.php

```diff
@@ -0,0 +1,98 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Security\Policy;
+
+use Neos\Flow\Annotations as Flow;
+
+/*
+* This file is part of the Neos.FluidAdaptor package.
+*
+* (c) Contributors of the Neos Project - www.neos.io
+*
+* This package is Open Source Software. For the full copyright and license
+* information, please view the LICENSE file which was distributed with this
+* source code.
+*/
+
+/**
+ * A collection class containing all roles for a account
+ *
+ * @Flow\Proxy(false)
+ */
+final class Roles implements \JsonSerializable, \IteratorAggregate, \Countable, \ArrayAccess
```

Comment:
I think this class should be immutable - otherwise this could lead to a real mess.
I.e. throw in `offsetSet` and `offsetUnset` and add methods like
```php
public function withRole(Role $role): self

public function withoutRole(string $roleIdentifier): self
```

or similar

Reply:
Changed like that, though I had `withoutRole(Role $role)`, but yeah, maybe makes sense to filter out roles with the plain role string too... though it also feels counter to the idea of having a VO in the first place. The use case **should be** s.th. like:

```
$editorRole = new Role('User.Editor'); // or have some static constructor even `MyRoles::Editor()`
...
$this->roles = $this->roles->withoutRole($editorRole);
```

**should**

Reply:
This is done in the Roles VO as I can see